datafusion/core/src/logical_plan/builder.rs

```diff
@@ -518,6 +518,18 @@ impl LogicalPlanBuilder {
         })))
     }
 
+    /// Apply an alias
+    pub fn alias(&self, alias: &str) -> Result<Self> {
```

Comment:
Ah, and I see you don't want to do that :) in the description

I wonder if you could use the `DFSchema::relation` name for this purpose?

Reply:
I think I need something to very specifically say that a table is being used as an alias. An easier change might be just to add an additional field to the `TableScan` to record the original table name. I think I will put up a separate PR for that approach.

So to explain why I need this. I want to use DataFusion as a SQL parser and planner but I want to execute the query in a different engine. I can provide a `TableProvider` so that DataFuision can get the schema for table `person` and I get a logical plan. When I go to translate that plan to a physical plan, it refers to a table called `peeps` (the alias) and I have no way to know the actual table name.

Reply:
This is the Spark plan for this use case:

```
'Project [*]
+- 'SubqueryAlias peeps
   +- 'UnresolvedRelation [person], [], false
```

Reply:
I think I prefer the approach taken in this PR, as it could handle subqueries rather than just table scans (as suggested by the name).

I think this is a cleaner, more generic approach.